### DISCUSSION TITLE
Global variables

### DISCUSSION THREAD

**Discussion by Radcriminal** [vectordotdev/vector#15172] [General]
I would like to know, does vector support global variables? I can create variable, for example in every remap transform, but it would be  great to create variable once and use it everywhere (at least within configuration file).

**Comment by jszwedko** [comment]
Hi @Radcriminal !

Not currently, but would you want to share more of your use case? We could open a feature request.

**Reply by Radcriminal** [reply]
I am working with alerting pipelines. I have multiple pipelines for different alert sources from different systems that in result sink into slack. I have a map with keys = severity (low, med, high) and values = emojis. I map severity of the incident to emoji and send it into slack. It's not a big deal, but I would like to create map globally and address to it in different transforms. Now, I need to create the same map in every transform stanza.

**Reply by spencergilbert** [reply]
I wonder if we could hack a workaround today with environment variables 🤔

**Comment by pchakour** [comment]
It will also great if we can reproduce the lua "Count the number of log" using remap transform.
Sometimes, we need to use a shared variable between events for aggregation purpose.

**Reply by ghost** [reply]
That approach would be very useful when we run on the VM/Cloud/Kubernetes and would like to use some metadata.

We can add a variables from a metadata endpoint once, to not perform a call every time:
```yaml
variables:
  AWS_ECS_CONTAINER_NAME = parse_url!(${ECS_CONTAINER_METADATA_URI_V4}).parsed.Containers.Name
```

It also would be useful if we would be able to use different sources/transforms inside the `variables`.
